src/sys/ptrace/linux.rs

```diff
@@ -213,6 +238,48 @@ pub fn getregs(pid: Pid) -> Result<user_regs_struct> {
     ptrace_get_data::<user_regs_struct>(Request::PTRACE_GETREGS, pid)
 }
 
+/// Get user registers, as with `ptrace(PTRACE_GETREGSET, pid, NT_PRSTATUS, ...)`
+#[cfg(all(
+    target_os = "linux",
+    target_env = "gnu",
+    any(
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+    )
+))]
+pub fn getregs(pid: Pid) -> Result<user_regs_struct> {
```

Comment:
I wonder can we implement this method using `PTRACE_GETREGS` on these non-x86 arches?

Reply:
> In fact, PTRACE_{GET,SET}REGS symbols are missing on aarch64 and riscv64, so we have to use PTRACE_{GET,SET}REGSET to achieve the same effect.

Hi, can we document this in the code (for `getregs` and `setregs`) so that developers who are not familiar with these APIs can understand why we are having 2 different implementations for the same API:)

Reply:
Done. :)